server/app/lib/actions/fusor/deploy.rb

```diff
@@ -0,0 +1,67 @@
+#
+# Copyright 2015 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public
+# License as published by the Free Software Foundation; either version
+# 2 of the License (GPLv2) or (at your option) any later version.
+# There is NO WARRANTY for this software, express or implied,
+# including the implied warranties of MERCHANTABILITY,
+# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
+# have received a copy of GPLv2 along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+
+module Actions
+  module Fusor
+    class Deploy < Actions::EntryAction
+      def humanized_name
+        _("Deploy")
+      end
+
+      def plan(deployment)
+        fail _("Unable to locate fusor.yaml settings in config/settings.plugins.d") unless SETTINGS[:fusor]
+
+        sequence do
+          # TODO: add an action to support importing a manifest created as part of the deployment
+
+          plan_action(::Actions::Fusor::Content::EnableRepositories, deployment)
+
+          # As part of enabling repositories, zero or more repos will be created.  Let's
+          # retrieve the repos needed for the deployment and use them in actions that follow
+          repositories = retrieve_deployment_repositories
+
+          plan_action(::Actions::Fusor::Content::SyncRepositories, repositories)
+          plan_action(::Actions::Fusor::Content::PublishContentView, deployment, repositories)
+        end
+      end
+
+      private
+
+      def retrieve_deployment_repositories
+        repos = []
+        if content = SETTINGS[:fusor][:content]
+          content[:rhev].each{ |details| repos << find_repository(details) } if content[:rhev]
+          content[:cloudforms].each{ |details| repos << find_repository(details) } if content[:cloudforms]
+          content[:openstack].each{ |details| repos << find_repository(details) } if content[:openstack]
+        end
```

Comment:
could be better to do `[content[:rhev], content[:cloudforms], content[:openstack]].compact.flatten(1).each { ... }` it has less duplications